tests/cephfs/cephfs_bugs/test_mds_dir_max_commit_size.py

```diff
@@ -0,0 +1,158 @@
+            cmd=f"python3 /home/cephuser/smallfile/smallfile_cli.py --operation create "
+            f"--threads 1 --file-size 1 --files 1048576 --top {fuse_mount_dir}",
+            timeout=10000,
+        )
+
+        # Perform cleanup after workload
+        client1.exec_command(
+            sudo=True, cmd=f"rm -rf {fuse_mount_dir}", check_ec=False, timeout=10000
+        )
+
+        # Ensure the monitor thread ends after the rm -rf operation
+        monitor_thread.do_run = False  # Signal to stop monitoring
+        monitor_thread.join()  # Wait for the monitoring thread to finish
+
+        return 0
+
+    except CommandFailed as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        client1.exec_command(sudo=True, cmd=f"umount {fuse_mount_dir}", check_ec=False)
+        client1.exec_command(
+            sudo=True, cmd="ceph config set mds mds_dir_max_commit_size 10"
+        )
+        client1.exec_command(
+            sudo=True, cmd="ceph config set mon mon_allow_pool_delete true"
+        )
+        fs_util.remove_fs(client1, fs_name)
+
+
+def monitor_mds_state(client1, fs_name, interval=5):
+    """
+    Monitors the MDS states and logs changes with a datetime stamp.
+    Runs in a separate thread and stops when signaled.
+
+    Parameters:
+    client1: Object to execute commands on the Ceph cluster.
+    interval (int): Time interval in seconds between health checks.
+    """
+    thread = threading.currentThread()
+    previous_mds_states = {}
+
+    while getattr(thread, "do_run", True):
+        try:
+            # Run command to get the current Ceph MDS state in JSON format
+            out, rc = client1.exec_command(
+                sudo=True, cmd=f"ceph fs status {fs_name} -f json"
+            )
+            current_data = json.loads(out)
+
+            current_mds_states = {}
+
+            # Extract current MDS states (name and state)
+            for mds in current_data["mdsmap"]:
+                name = mds["name"]
+                state = mds["state"]
+                current_mds_states[name] = state
+
+            log.info(f"Current MDS states: {current_mds_states}")
+
+            # Check for changes in the MDS states
+            for mds_name, current_state in current_mds_states.items():
+                previous_state = previous_mds_states.get(mds_name)
+
+                if previous_state is None:
+                    previous_mds_states[mds_name] = current_state
+                elif current_state != previous_state:
+                    log.info(f"{datetime.now()} - MDS state changed!\n")
+                    log.info(f"MDS: {mds_name}\n")
+                    log.info(f"Previous state: {previous_state}\n")
+                    log.info(f"Current state: {current_state}\n")
+                    previous_mds_states[mds_name] = current_state
+
```

Comment:
Is there any unexpected MDS state transition we want to track here? Because we are noting all transitions and printing in log, which may not look into unless we raise soem assertion or failure. So all transitions are expected?